### PR TITLE
Fix usage of "package" tag in file headers

### PR DESCRIPTION
**Pull request by claudiosanches**
This PR fixes the usage of `@package` tag, it will improve the code reference generated from our code base.

Note that now inside `src/` isn't required a file comment or `@package` since those classes use namespaces.

We have already fixed those on core in https://github.com/woocommerce/woocommerce/pull/27239
Those changes will fix "Packages" section in the sidebar of https://docs.woocommerce.com/wc-apidocs/

### CODE REVIEW

File: phpcs.xml.dist [hunk 1]
```diff
@@ -44,4 +44,12 @@
 	<rule ref="Generic.Commenting">
 		<exclude-pattern>tests/</exclude-pattern>
 	</rule>
+	<rule ref="Squiz.Commenting.FileComment.MissingPackageTag">
```
Comment (timmyc) [inline phpcs.xml.dist:47]: Thanks for adding these rules @claudiosanches

**Comment by claudiosanches** [comment]
@timmyc I just fixed all coding standards issues that I found, all tests are passing now.

**Comment by timmyc** [comment]
Cherry picked into `release/1.4.0` in d664157e00dd12fed18ed069fa14a4abbab1dc16

**Comment by becdetat** [comment]
Cherry picked into `release/1.5.0-beta.2` in 307372eec7e4d81443cff6b2eee12cfb615e3020

**Pull request merged**